### ISSUE TITLE
abort trap: 6 on exa -l --git with a directory that has a symlink to a non-existant file

### ISSUE THREAD

**Issue by jcolson** [ogham/exa#794]
If exa does something unexpected, or its output looks wrong, or it displays an error on the screen, or if it outright crashes, then please include the following information in your report:

- The version of exa being used (`exa --version`)
`exa v0.9.0`
- The command-line arguments you are using
`exa -l --git`
- Your operating system and hardware platform
`Darwin macbook-air.lan 20.1.0 Darwin Kernel Version 20.1.0: Sat Oct 31 00:07:10 PDT 2020; root:xnu-7195.50.7~2/RELEASE_ARM64_T8101 arm64`

If it’s a crash, please include the full text of the crash that gets printed to the screen. If you’re seeing unexpected behaviour, a screenshot of the issue will help a lot.

---
```sh
~  src  …  personal  home  bin  $    exa -l --git
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 2, kind: NotFound, message: "No such file or directory" }', src/fs/feature/git.rs:271:22
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
Abort trap: 6
```

this happens when there is a symlink that points to a non existent file

**Comment by missinglink** [comment]
same here:

```
exa v0.9.0

exa -la --git --group-directories-first --icons --color-scale

Darwin Peters-MacBook-Pro-2.local 19.6.0 Darwin Kernel Version 19.6.0: Thu Jun 18 20:49:00 PDT 2020; root:xnu-6153.141.1~1/RELEASE_X86_64 x86_64
```

```
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 2, kind: NotFound, message: "No such file or directory" }', src/fs/feature/git.rs:271:22
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
[1]    52242 abort      exa -la --git --group-directories-first --icons --color-scale
```

symlink in the CWD:

```
test_cases -> ./acceptance-tests/test_cases/
```

the path `./acceptance-tests` exists but `./acceptance-tests/test_cases/` does not.

```
ls -lah acceptance-tests
total 0
drwxr-xr-x  2 peter  staff    64B Nov  4  2019 .
drwxr-xr-x  7 peter  staff   224B Jan 19 13:17 ..
```

removing the symlink allowed `exa` to function without error when executing the same command

**Comment by ariasuni** [comment]
This issue is the same as #526, and that it has been fixed in #584, there just hasn’t been a new release made since then.
